### ISSUE TITLE
[MVUX/CSM] Resources don't update after changing Fluent Theme manually 

### ISSUE THREAD

**Issue by morning4coffe-dev** [unoplatform/uno.extensions#2429]
Changing the theme manually with a ToggleSwitch inside the app results in the app not updating its background on Windows and on other platforms, the color is too dark when in DarkTheme. 

Tested on my side on `"Uno.Sdk": "5.2.175"` 

Repro with my notes: https://github.com/morning4coffe-dev/uno-platform-testing/tree/bug/mvux-csm-fluent-theme-resource-update

Then testing the same case in XAML+MVVM project, I am not able to repro this issue. 

_Originally posted by @kucint in https://github.com/unoplatform/uno/discussions/17487_:

Changing Theme to dark and back to light works fine on `Desktop`.
On `Windows`, controls seem to react correctly on change, except the main frame background which stays always light.

```cs
    public MainPage() => this
        .DataContext(new BindableMainPageModel(), (page, vm) => page
        .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))   // -> does not work on windows
        .Content(
            new ToggleSwitch()
            .OnContent("Dark mode")
            .OffContent("Light mode")
            .Name(out ToggleSwitch c, c => c.Toggled += (s, e) =>
            {
                FrameworkElement? content = XamlRoot?.Content as FrameworkElement;
                if (content is not null)
                    content.RequestedTheme = content.RequestedTheme == ElementTheme.Dark ? ElementTheme.Light : ElementTheme.Dark;
            })
            .IsOn(x => x.Binding(() => vm.IsAppInDarkMode).TwoWay())
        ));
```

Model:
```cs
internal partial record MainPageModel
{
    public IState<bool> IsAppInDarkMode => State<bool>.Value(this, () => false);
}
```
![image](https://github.com/user-attachments/assets/ac1640be-c19d-4a5c-ae41-198cd898ee59)

![image](https://github.com/user-attachments/assets/587a6717-0780-4cf4-b916-dc79c9d9f31e)

> [!NOTE]
> Latest dev branch used: `"Uno.Sdk": "5.4.0-dev.99"`

**Comment by agneszitte** [comment]
@kazo0 to verify if it is a duplicate: https://github.com/unoplatform/uno.csharpmarkup/issues/716

**Comment by kazo0** [comment]
@agneszitte yes this is the same issue as https://github.com/unoplatform/uno.csharpmarkup/issues/716

**Comment by MartinZikmund** [comment]
To work around this issue, you can create a `Style` targeting `Page`, with a setter for the `Background` (using `ThemeResource` in the `Setter`) - theme resources should be correctly applied and observed when part of `Style`, and this should work across all targets including WinUI

**Comment by MartinZikmund** [comment]
Workaround similar to https://github.com/microsoft/microsoft-ui-xaml/issues/9464#issuecomment-2258432746

**Comment by MartinZikmund** [comment]
Created https://github.com/microsoft/microsoft-ui-xaml/issues/10029
